samples/scanner/scanners/firewall_rules/whitelist.yaml

```diff
@@ -0,0 +1,133 @@
+#     See: https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy
+#   c. Return the first applicable rules.
+# 3. For each applicable rule, check if the policy is a superset of one of the "match_policies" defined in the rule.
+# 4. For any rule with a matching policy, check if the policy is a subset of any of the "verify_policies"
+#     if not, create a RuleViolation.
+
+rules:
+
+    # This rule will trigger on policies that allow anything outside of 10.0.0.0/8.
+    #
+    # Example triggering firewall rule:
+    #  {
+    #    "name": "firewall1",
+    #    "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #    "priority": 1000,
+    #    "sourceRanges": [
+    #      "10.0.0.0/7",
+    #      ],
+    #    "targetTags": [
+    #     "test",
+    #     ],
+    #    "allowed": [
+    #      {
+    #        "IPProtocol": "tcp",
+    #        "ports": [
+    #          "3389"
+    #        ]
+    #      }
+    #    ],
+    #  }
+    #
+    # Example firewall rule that WON'T trigger:
+    #
+    #  {
+    #    "name": "firewall2",
+    #    "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #    "priority": 1000,
+    #    "sourceRanges": [
+    #      "10.0.0.0/8",
+    #      ],
+    #    "targetTags": [
+    #     "test",
+    #     ],
+    #    "allowed": [
+    #      {
+    #        "IPProtocol": "all",
+    #      }
+    #    ],
+    #  }
+  - rule_id: 'test_instances_rule'
+    description: Test instances can only be exposed to corp IPs
+    mode: whitelist
+    match_policies:
+      - direction: ingress
+        allowed: ['*']
+        targetTags:
+          - 'test'
+    verify_policies:
+      - allowed: ['*']
+        sourceRanges:
+        - '10.0.0.0/8'
+
+    # This rule will trigger on policies that expose ports to 0.0.0.0/0 except for 80/443.
+    #
+    # Example triggering firewall rule:
+    #  {
+    #    "name": "firewall1",
+    #    "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #    "priority": 1000,
+    #    "sourceRanges": [
+    #      "0.0.0.0/0",
+    #      ],
+    #    "allowed": [
+    #      {
+    #        "IPProtocol": "tcp",
+    #        "ports": [
+    #          "22",
+    #          "80",
+    #          "443",
+    #        ]
+    #      }
+    #    ],
+    #  }
+    #
+    # Example firewall rule that WON'T trigger:
+    #
+    #  {
+    #    "name": "firewall2",
+    #    "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #    "priority": 1000,
+    #    "sourceRanges": [
+    #      "0.0.0.0/1",
+    #      ],
+    #    "allowed": [
+    #      {
+    #        "IPProtocol": "tcp",
+    #        "ports": [
+    #          "22",
+    #          "80",
+    #          "443",
+    #        ]
+    #      }
+    #    ],
+    #  }
+  - rule_id: 'https_allowed'
```

Comment:
nit: this rule naming should also include ```http``` otherwise, it seems the rule should not include port 80: ```alllow_http_and_https```